src/game/common/thing/thingtemplate.cpp

```diff
@@ -297,8 +466,364 @@ void ThingTemplate::Resolve_Names()
+    m_assetScale = 1.0f;
+    m_instanceScaleFuzziness = 0.0f;
+    m_structureRubbleHeight = 0;
+    m_displayName.Translate(name);
+    m_shadowType = SHADOW_VOLUME;
+    m_geometryInfo.Set(GEOMETRY_SPHERE, false, 10.0f, 10.0f, 10.0f);
+}
+
+FieldParse *ThingTemplate::Get_Field_Parse()
+{
+    return s_objectFieldParseTable;
+}
+
+FieldParse *ThingTemplate::Get_Reskin_Field_Parse()
+{
+    return s_objectReskinFieldParseTable;
+}
+
+void ThingTemplate::Parse_Add_Module(INI *ini, void *instance, void *store, const void *user_data)
+{
+    ThingTemplate *tmplate = (ThingTemplate *)instance;
+    unsigned char oldState = tmplate->m_moduleParseState;
+
+    if (oldState != MODULEPARSE_NORMAL) {
+        throw CODE_06;
+    }
+
+    tmplate->m_moduleParseState = MODULEPARSE_MODIFY_ON_COPY;
+    ini->Init_From_INI(tmplate, Get_Field_Parse());
+    tmplate->m_moduleParseState = oldState;
+}
+
+void ThingTemplate::Parse_Armor_Template_Set(INI *ini, void *instance, void *store, const void *user_data)
+{
+    // todo needs more of ArmorTemplateSet and SparseMatchFinder
+#ifdef GAME_DLL
+    Call_Function<void, INI *, void *, void *, const void *>(
+        PICK_ADDRESS(0x0058A830, 0x006EA8DB), ini, instance, store, user_data);
+#endif
+}
+
+void ThingTemplate::Parse_Inheritable_Module(INI *ini, void *instance, void *store, const void *user_data)
+{
+    ThingTemplate *tmplate = (ThingTemplate *)instance;
+    unsigned char oldState = tmplate->m_moduleParseState;
+
+    if (oldState != MODULEPARSE_NORMAL) {
+        throw CODE_06;
+    }
+
+    tmplate->m_moduleParseState = MODULEPARSE_KEEP_ON_COPY;
+    ini->Init_From_INI(tmplate, Get_Field_Parse());
+    tmplate->m_moduleParseState = oldState;
+}
+
+void ThingTemplate::Parse_Int_List(INI *ini, void *formal, void *store, const void *user_data)
+{
+    for (int i = 0; i < reinterpret_cast<intptr_t>(user_data); i++) {
+        static_cast<int *>(store)[i] = INI::Scan_Int(ini->Get_Next_Token());
+    }
+}
+
+void ThingTemplate::Parse_Max_Simultaneous(INI *ini, void *instance, void *store, const void *user_data)
+{
+    char str[36];
+    strcpy(str, "DeterminedBySuperweaponRestriction");
+    ThingTemplate *tmplate = (ThingTemplate *)instance;
+
+    captainslog_dbgassert(&tmplate->m_maxSimultaneousOfType == store, "Bad store passed to parseMaxSimultaneous");
+
+    const char *str2 = ini->Get_Next_Token();
+
+    if (!strcasecmp(str2, str)) {
+        tmplate->m_determinedBySuperweaponRestriction = true;
+        *static_cast<uint16_t *>(store) = 0;
+    } else {
+        int tmp = ini->Scan_Int(str2);
+
+        captainslog_relassert(tmp >= 0 && tmp < 65535, 0xDEAD0001, "Value parsed outside range of a short.");
+
+        *static_cast<uint16_t *>(store) = tmp;
+        tmplate->m_determinedBySuperweaponRestriction = false;
+    }
+}
+
+bool ThingTemplate::Remove_Module_Info(const Utf8String &tag, Utf8String &name)
+{
+    bool ret = m_body.Clear_Module_Data_With_Tag(tag, name);
```

Comment:
lets keep it in a if like the rest for easier reading

Reply:
fixed.